fdbserver/ClusterController.actor.cpp

```diff
@@ -223,6 +223,243 @@ bool ClusterControllerData::remoteTransactionSystemContainsDegradedServers() {
 	return false;
 }
 
+ACTOR Future<Void> recruitSingleLogRouter(ClusterControllerData* cluster,
+                                          ClusterControllerData::DBInfo* db,
+                                          int logRouterTagId,
+                                          int logSetIndex,
+                                          Reference<ILogSystem> logSystem,
+                                          LogSystemConfig config) {
+	state Tag routerTag = Tag(tagLocalityLogRouter, logRouterTagId);
+
+	TraceEvent("RecruitingLogRouter", cluster->id)
+	    .detail("TagId", logRouterTagId)
+	    .detail("LogSetIndex", logSetIndex)
+	    .detail("Tag", routerTag.toString());
+
+	// Step 1: Determine start version
+	// The peek cursor will automatically adjust to the actual popped version on TLogs,
+	// so we can just start at version 0
+	state Version startVersion = 0;
+
+	TraceEvent("LogRouterStartVersion", cluster->id)
+	    .detail("TagId", logRouterTagId)
+	    .detail("StartVersion", startVersion)
+	    .detail("Note", "PeekCursorWillAdjustToPopperVersion");
+
+	// Step 2: Recruit a worker for the new log router
+	state Optional<Key> targetDcId;
+	if (config.tLogs[logSetIndex].locality == tagLocalitySatellite) {
+		for (auto& region : db->config.regions) {
+			for (auto& dc : region.satellites) {
+				targetDcId = dc.dcId;
```

Comment:
If there are multiple satellites is it fine to pick the dcId of any of the satellites here? Or does the dcId need to be provided as an input to this actor?

Reply:
I think you can ignore this comment.